keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHierarchicalRegression.java

```diff
@@ -83,23 +74,18 @@ private RegressionModel linearRegression(List<Data> data) {
         return model;
     }
 
-    private void buildAndRunHeirarchicalNetwork(List<Data> radonData, int numberOfModels) {
+    private void buildAndRunHierarchicalNetwork(Map<String, List<Data>> radonData, int numberOfModels) {
         GaussianVertex muAlpha = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuIntercept");
         GaussianVertex muBeta = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuGradient");
 
         HalfGaussianVertex sigmaAlpha = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaIntercept");
         HalfGaussianVertex sigmaBeta = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaGradient");
 
-        int numPartitions = radonData.size() / numberOfModels;
-
-        double[] allFloor = radonData.stream().mapToDouble(k -> k.floor).toArray();
-        double[] allRadon = radonData.stream().mapToDouble(k -> k.log_radon).toArray();
-
-        List<RegressionModel> models = new ArrayList<>();
+        List<String> counties = radonData.keySet().stream().sorted().collect(Collectors.toList());
 
-        for (int i = 0; i < numberOfModels; i++) {
-            RegressionModel model = createSubModel(allFloor, allRadon, i, numPartitions, muBeta, muAlpha, sigmaBeta, sigmaAlpha);
-            models.add(model);
+        for (int i = 0; i < Math.min(numberOfModels, counties.size()); i++) {
+            List<Data> countyData = radonData.get(counties.get(i));
+            createSubModel(countyData, muBeta, muAlpha, sigmaBeta, sigmaAlpha);
         }
 
         //Set the starting values of the parent random variables
```

Comment:
I've had to forcefully set these values near the correct answer to "help" the optimiser find the correct values. This is basically cheating. Can you check that it can find the correct values without doing the `setValue`?

Reply:
I've swapped out the hardcoded initialisation for inferred params for the single linear regression model